File: atividade-a1/src/grafo.cc
#include "grafo.h"
#include <fstream>
#include <string>
#include <iostream>


using namespace std;

Grafo::Grafo() {
    _vertices = 0;
    _arestas = 0;
    // _matriz = nullptr;
}

int Grafo::qtdVertices() {
    return _vertices;
}

int Grafo::qtdArestas() {
    return _arestas;
}

// lista
int grau(Nodo v) {

}

// char Grafo::rotulo(Nodo v) {
//     return v.rotulo();
// }

// lista
int vizinhos(Nodo v) {

}

// matriz
bool haAresta(Nodo v, Nodo u) {
    
}

// matriz
int peso(Nodo v, Nodo u) {
    
}

void Grafo::ler(string nome_arquivo) {
    fstream arquivo;
    arquivo.open(nome_arquivo);

    string conteudo;
    int linha = 0;
    string conteudo_linha;

    while (!arquivo.eof()) {
        linha++;
        getline(arquivo, conteudo_linha);
        // cout << "ola" + conteudo_linha + "\n";
        string tag = conteudo_linha;
        cout << tag + "\n";
    }

    arquivo.close();
}

File: atividade-a1/python/grafo.py
from vertice import Vertice

class Grafo:

    def __init__(self):
        self.__qtdVertices = 0
        self.__qtdArestas = 0
        self.__vertices = []
        self.__matriz = []


    def peso(self, vertice1, vertice2):
        pass

    def vizinhos(self, value):
        for vertice in self.__vertices:
            if vertice.id == value:
                return list(vertice.vizinhos.keys())

    def haAresta(self, vertice1, vertice2):
        pass

    def grau(self, vertice):
        pass

    def ler(self, txt):
        arquivo = open(txt, 'r')
        primeira_linha = arquivo.readline().split()
        for i in range(int(primeira_linha[1])):
            linha = arquivo.readline()
            linha = linha.split()
            print(linha)
            
            
            nome = ""
            for j in range(1, len(linha)-1):
                nome = nome + linha[j]
                nome = nome + ' '

            
            nome = nome + linha[len(linha)-1]
            nome = nome.replace('"', "")
            self.__vertices.append(Vertice(int(linha[0]), nome))
        arquivo.readline()
        for linha in arquivo:
            linha = linha.split()
            vert_u = int(linha[0])
            vert_v = int(linha[1])
            weight_u_v = float(linha[2])
            self.__vertices[vert_u].vizinhos[vert_v] = weight_u_v
            self.__vertices[vert_v].vizinhos[vert_u] = weight_u_v
        arquivo.close()

    @property
    def qtdVertices(self):
        return self.__qtdVertices

    @property
    def qtdArestas(self):
        return self.__qtdArestas
    
    @property
    def vertices(self):
        return self.__vertices